sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -576,67 +563,95 @@ private void OverrideBindingName(INamedTypeSymbol attributeClass, ref string arg
             /// attribute must be an enumerable type.
             /// </summary>
-            private bool IsEventHubsTriggerValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType, out Cardinality cardinality)
+            private bool IsCardinalityValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType)
             {
                 dataType = DataType.Undefined;
-                cardinality = Cardinality.Undefined;
+                var cardinalityIsNamedArg = false;
 
                 // check if IsBatched is defined in the NamedArguments
                 foreach (var arg in attribute.NamedArguments)
                 {
                     if (String.Equals(arg.Key, Constants.FunctionMetadataBindingProps.IsBatchedKey) &&
                         arg.Value.Value != null)
                     {
+                        cardinalityIsNamedArg = true;
                         var isBatched = (bool)arg.Value.Value; // isBatched takes in booleans so we can just type cast it here to use
 
                         if (!isBatched)
                         {
                             dataType = GetDataType(parameterSymbol.Type);
-                            cardinality = Cardinality.One;
                             return true;
                         }
                     }
                 }
 
-                // Check the default value of IsBatched
-                var eventHubsAttr = attribute.AttributeClass;
-                var isBatchedProp = eventHubsAttr!.GetMembers().Where(m => string.Equals(m.Name, Constants.FunctionMetadataBindingProps.IsBatchedKey, StringComparison.OrdinalIgnoreCase)).SingleOrDefault();
-                AttributeData defaultValAttr = isBatchedProp.GetAttributes().Where(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, Compilation.GetTypeByMetadataName(Constants.Types.DefaultValue))).SingleOrDefault();
-                var defaultVal = defaultValAttr.ConstructorArguments.SingleOrDefault().Value!.ToString(); // there is only one constructor arg, the default value
-                if (!bool.TryParse(defaultVal, out bool b) || !b)
+                // When "IsBatched" is not a named arg, we have to check the default value
+                if (!cardinalityIsNamedArg)
                 {
-                    dataType = GetDataType(parameterSymbol.Type);
-                    cardinality = Cardinality.One;
-                    return true;
+                    var attrClass = attribute.AttributeClass;
+                    var isBatchedProp = attrClass!
+                        .GetMembers()
+                        .SingleOrDefault(m => string.Equals(m.Name, Constants.FunctionMetadataBindingProps.IsBatchedKey, StringComparison.OrdinalIgnoreCase));
```

Comment:
This code block exists a couple times I think? Might be worth it's own private method